go/enclave/db/storage.go

```diff
@@ -199,6 +199,11 @@ func (s *storageImpl) IsBlockAncestor(block *types.Block, maybeAncestor common.L
 	return s.IsBlockAncestor(p, maybeAncestor)
 }
 
+func (s *storageImpl) HealthCheck() (bool, error) {
+	headRollup := s.FetchHeadRollup()
```

Comment:
@otherview I've checked and `FetchHeadRollup` calls `ReadHeadRollupHash`, which actually swallows any errors. Can you change that method to actually return an error? Then I think he check here should be whether an error is returned, not whether the head rollup is nil (otherwise, there's the edge case of the healthcheck being performed too early).

Reply:
That's *one* reason why I left the healthcheck simple. The Storage is swallowing a bunch of errors ( same thing in the StoreRollup ). I'm not sure why we're not surfacing errors, maybe we were panicking before and it was all in mem  ? 

I can do that in a new PR  though. Maybe changing the error handling on the storage package ?

Reply:
I guess so. It just seems dumb to have a healthcheck that pings the DB, except that it swallows any errors (I get that it works here, assuming a first rollup has been produced). Why don't we add a todo for now?

Reply:
Agreed. Let me add a todo 👍